AoE2ScenarioParser/datasets/projectiles.py

```diff
@@ -42,16 +98,22 @@ def get_unit_projectile(unit_const: int, has_chemistry: bool = False, secondary:
         Returns:
             The ProjectileInfo corresponding to the given unit and params or None if nothing was found
         """
+
         if type(unit_const) is not int:
             raise TypeError(f"unit const expected int, got {type(unit_const)}")
-        if unit_const == -1:
-            raise ValueError("-1 is not a valid unit constant")
+        if unit_const < 0:
+            raise ValueError(f"{unit_const} is not a valid unit constant")
 
         projectile = "secondary" if secondary else "primary"
         fire = "_fire" if has_chemistry else ""
         projectile_id = _unit_projectile_info.get(str(unit_const), {}).get(projectile + fire, -1)
 
-        return ProjectileInfo.from_id(projectile_id) if projectile_id else None
+        if projectile_id != -1:
+            return ProjectileInfo.from_id(projectile_id)
+
+        raise KeyError(
+            f"The unit with ID '{unit_const}' does not use a {projectile} projectile{' with fire' if fire else ''}"
+        )
```

Comment:
I don't think raising an error is a good idea here. As requesting a projectile based on a unit will result in errors even though the user didn't do anything wrong. I personally when programming rather check the return type being 0 than try excepting this

Reply:
Fixing this in the new commit